### PR TITLE
[SPARK-29825][SQL][TESTS] Add join-related configs in `inner-join.sql` and `postgreSQL/join.sql`

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

For better test coverage, this pr is to add join-related configs in `inner-join.sql` and `postgreSQL/join.sql`. These join related configs were just copied from ones in the other join-related tests in `SQLQueryTestSuite` (e.g., https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/natural-join.sql#L2-L4).

### Why are the changes needed?

Better test coverage.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by maropu** [comment]
cc: @wangyum @dongjoon-hyun @HyukjinKwon

**Comment by dongjoon-hyun** [comment]
Hi, @maropu . This PR adds comment. Is `Add join conditions in join-related tests` correct?

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql [hunk 1]
```diff
@@ -6,6 +6,11 @@
 -- Test JOIN clauses
 -- https://github.com/postgres/postgres/blob/REL_12_BETA2/src/test/regress/sql/join.sql
 --
+
+--SET spark.sql.autoBroadcastJoinThreshold=10485760
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql]: Also, this looks like a magic number, `10485760`. Could you explain why you choose this value? I don't see this kind of value from the original `https://github.com/postgres/postgres/blob/REL_12_BETA2/src/test/regress/sql/join.sql`.

Comment (maropu) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql]: I just copied them from the other join-related tests (e.g., https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/natural-join.sql#L2-L4), so I'm not sure about why the value chosen. I think this configuration is just to prohibit broadcast hash joins for tests.

**Comment by maropu** [comment]
> Hi, @maropu . This PR adds comment. Is Add join conditions in join-related tests correct?

Ambiguous? I updated it.

**Comment by dongjoon-hyun** [comment]
Shall we add the following line as a first line? This PR is adding a comment describing a running environment instead of adding the real configration. I mean that's confusing.
```
-- List of configuration the test suite is run against:
```

**Comment by maropu** [comment]
Ah, ok.

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql (hunk 1 above)
Comment (AngersZhuuuu) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql]: @dongjoon-hyun @maropu 
 `10 * 1024 * 1024 = 10485760` 
This config's default value is `10m`

Comment (mgaido91) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql]: right, it is the default value

Comment (maropu) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/join.sql]: Thanks, @mgaido91 !

**Review by wangyum — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
Thank you, @maropu , @mgaido91 , @AngersZhuuuu , @wangyum .

**Pull request closed without merging**